controllers/add_pods.go

```diff
@@ -245,6 +250,32 @@ func createInstance(r *FoundationDBClusterReconciler, context ctx.Context, clust
 			err = r.Create(context, service)
 			return false, err
 		}
+
+		// If we create a new Pod and reuse a service we should ensure that it matches.
+		if processClass == fdbtypes.ProcessClassStorage {
+			// Ensure that the service contains all required ports.
+			// Since a process can listen on non-TLS and TLS we multiply by 2.
+			// Since we are creating a new Pod we can use the cluster.GetStorageServersPerPod()
+			// if this value would be changed during the creation it should be reconciled again.
+			if len(services.Items[0].Spec.Ports) != (cluster.GetStorageServersPerPod() * 2) {
+				log.Info("Updating service missing ports", "got", len(services.Items[0].Spec.Ports), "expected", cluster.GetStorageServersPerPod()*2)
+				log.Info("Updating service missing ports", "namespace", cluster.Namespace, "cluster", cluster.Name, "serviceName", services.Items[0].ObjectMeta.Name)
```

Comment:
Should we combine these into a single log item? It may be harder to search through logs with two separate items.

Reply:
Makes sense 👍